src/Orc.Controls/Orc.Controls.Shared/Controls/DateTimePicker/Views/DateTimePickerControl.xaml.cs

```diff
@@ -245,6 +294,209 @@ public override void OnApplyTemplate()
             base.OnApplyTemplate();
 
             AccentColorBrush = TryFindResource("AccentColorBrush") as SolidColorBrush;
+
+            EnableOrDisableHourConverterDependingOnFormat();
+        }
+
+        private void OnFormatChanged()
+        {
+            ApplyFormat();
+        }
+
+        private void ApplyFormat()
```

Comment:
Well, my idea is to create FormatHelper which will return FormatInfo object containing all information needed for UI. Then we can just write unit test that will pass different formats and check whether resulting object is corrent. How about this ?